ts/nni_manager/training_service/reusable/environments/dlcEnvironmentService.ts

```diff
@@ -86,6 +94,8 @@ export class DlcEnvironmentService extends EnvironmentService {
                     environment.setStatus('UNKNOWN');
             }
         });
+        deferred.resolve();
```

Comment:
This line will be executed before line 87.
If you want to resolve the promise after forEach finish, you should use `await Promise.all(environments.map(...))`.

Reply:
it's ok
I don't need to reslove ,because it just update dlc job status